FiftyOne.Pipeline.Core/FlowElements/Pipeline.cs

```diff
@@ -453,24 +486,36 @@ private void AddElementsByType(IReadOnlyList<IFlowElement> elements)
 
         private void AddAvailableProperties(
             IReadOnlyList<IFlowElement> elements,
-            IDictionary<string, IDictionary<string, IElementPropertyMetaData>> dictionary)
+            IDictionary<string, IDictionary<string, IElementPropertyMetaData>> dictionary,
+            ref bool hadFailures)
         {
             foreach (var element in elements)
             {
                 if (element is ParallelElements)
                 {
-                    AddAvailableProperties(((ParallelElements)element).FlowElements, dictionary);
+                    AddAvailableProperties(((ParallelElements)element).FlowElements, dictionary, ref hadFailures);
                 }
                 else
                 {
+                    IList<IElementPropertyMetaData> elementProps;
+                    try
+                    {
+                        elementProps = element.Properties;
+                    }
+                    catch (PipelineException)

```

Comment:
can we guarantee that if the exception is thrown it is only of type `PipelineException`?  also maybe worth to check for null?  so that we don't crash lower if elementProps is null?

Reply:
`FlowElementBase` is implemented as:
```
        public abstract IList<TMeta> Properties { get; }

        /// ...
        IList<IElementPropertyMetaData> IFlowElement.Properties
        {
            get
            {
                return Properties.Cast<IElementPropertyMetaData>().ToList();
            }
        }
```
If any derived instance returned `null` from the overridden (no longer abstract) property, this implementation would throw [ArgumentNullException](https://learn.microsoft.com/en-us/dotnet/api/system.argumentnullexception?view=netframework-4.6.2) as per specification of Linq [`ToList` extension method](https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.tolist?view=netframework-4.6.2).

All derived classes override the new abstract property in a way that returns an instance created during either the overridden getter or the constructor (of that specific derived class), see [[search results](https://github.com/search?q=repo%3Apostindustria-tech%2Fpipeline-dotnet+%22override+IList%3CIElementPropertyMetaData%3E+Properties%22&type=code)].

The only 2 exceptions to his convention are:
- [`ParallelElements` always throws `NotImplementedException`](https://github.com/postindustria-tech/pipeline-dotnet/blob/6bf5194320e7fe24eef069e92166aafd59029878/FiftyOne.Pipeline.Core/FlowElements/ParallelElements.cs#L99) _(but is explicitly handled in separate branch)_
- [`CloudAspectEngineBase` may throw `PipelineException`](https://github.com/postindustria-tech/pipeline-dotnet/blob/main/FiftyOne.Pipeline.CloudRequestEngine/FlowElements/CloudAspectEngineBase.cs#L175)

Reply:
Updated `catch` clause to only capture new `PropertiesNotYetLoadedException` to prevent interference with other critical conditions, e.g. `PipelineConfigurationException`. Added wrapping logic to `CloudAspectEngineBase.Properties`.

Reply:
makes sense, thank you.